packages/design-system/src/components/dropDown/index.js

```diff
@@ -85,12 +86,34 @@ export const DropDown = forwardRef(
     ref
   ) => {
     const internalRef = useRef();
+    const [dynamicPlacement, setDynamicPlacement] = useState(placement);
 
     const { activeOption, isOpen, normalizedOptions } = useDropDown({
       options,
       selectedValue,
     });
 
+    const positionPlacement = useCallback(
+      (popupRef) => {
+        // check to see if there's an overlap with the window edge
+        const dimensions = popupRef.current?.getBoundingClientRect();
+        const bottom = dimensions?.bottom;
+        const top = dimensions?.top;
```

Comment:
🤔  nit: 

```suggestion
        const { bottom, top } = popupRef.current?.getBoundingClientRect();
```

Reply:
Or I guess I could give it a fallback 🤦🏻‍♀️